puffin_egui/src/flamegraph.rs

```diff
@@ -665,13 +679,21 @@ fn paint_record(
     if wide_enough_for_text {
         let painter = info.painter.with_clip_rect(rect.intersect(info.canvas));
 
-        let duration_ms = to_ms(record.duration_ns);
-        let text = if record.data.is_empty() {
-            format!("{}{} {:6.3} ms {}", prefix, record.id, duration_ms, suffix)
+        let scope_details = info.scope_collection.fetch_by_id(&scope_id).unwrap();
+
+        let scope_type = match scope_details.scope_type() {
+            ScopeType::Function(name) | ScopeType::Scope(name) => name,
+        };
```

Comment:
build this in to the `ScopeType` instead? 🤷 

```suggestion
        let scope_type = match scope_details.name();
```